src/main.rs

```diff
@@ -3,49 +3,84 @@ use env_logger::Env;
 use log::{debug, info, warn};
 
 use crate::compiler::{compile_configuration, load_file};
+use crate::expectation::model::display_expectations;
 use crate::web::State;
 use std::sync::{Arc, RwLock};
+use std::time::Instant;
+use tokio::sync::mpsc::{channel, Receiver};
+use tokio::task::spawn_blocking;
 
 mod cli;
 mod compiler;
 mod expectation;
 mod mock;
 mod web;
 
-#[tokio::main]
-async fn main() -> Result<(), Error> {
+fn main() -> Result<(), Error> {
     start_logger();
 
     let cli_args = cli::load_cli_args();
 
-    info!("Start dhall mock project 👋");
+    let loading_rt = tokio::runtime::Builder::new()
+        .threaded_scheduler()
+        .thread_stack_size(8 * 1024 * 1024)
+        .core_threads(1)
+        .max_threads(3)
+        .build()?;
 
-    let expectations = cli_args
-        .configuration_files
-        .iter()
-        .flat_map(|configuration_file| {
-            debug!("Loading configuration file {}", configuration_file);
-            let configuration_result = load_file(configuration_file)
-                .and_then(|configuration| compile_configuration(&configuration))
-                .context("Error loading configuration");
-            match configuration_result {
-                Ok(expectations) => expectations.into_iter(),
-                Err(e) => {
-                    warn!("{}", e);
-                    Vec::new().into_iter()
-                }
-            }
-        })
-        .collect();
-    info!("Loaded expectations : {:?}", expectations);
+    let mut web_rt = tokio::runtime::Runtime::new()?;
+
+    info!("Start dhall mock project 👋");
+    let (mut tx, rx) = channel(10);
+    let state = Arc::new(RwLock::new(State {
+        expectations: vec![],
+    }));
 
-    let state = Arc::new(RwLock::new(State { expectations }));
-    web::web_server(state.clone(), cli_args.http_bind).await?;
+    let compiler_state = state.clone();
+    loading_rt.spawn(compiler_executor(rx, compiler_state));
 
-    Ok(())
+    let web_state = state.clone();
+    web_rt.block_on(async move {
```

Comment:
Why couldn't we do the code process below directly inside the main?

Reply:
Because `tx.send(configuration.clone()).await?` come from `tokio` channels that need to be in an async environment.
Initialy I tried to add it in the main but it should be spawned and split the tasks change the borrowing of `cli_args`

Reply:
LGTM